### PR TITLE
Add test for documentation build

### PR DESCRIPTION
**Pull request by leckie-chn**
add test for sphinx using `-W` (Warning-as-Error)
fix all warnings to pass test.

### CODE REVIEW

File: docs/en_US/Compressor/Quantizer.md [hunk 1]
```diff
@@ -6,10 +6,12 @@ We provide Naive Quantizer to quantizer weight to default 8 bits, you can use it
 
 ### Usage
 tensorflow
-```python nni.compression.tensorflow.NaiveQuantizer(model_graph).compress()
+```
+python nni.compression.tensorflow.NaiveQuantizer(model_graph).compress()
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Quantizer.md]: tensorflow section should be removed for now.

pytorch
```
```python
model = nni.compression.torch.NaiveQuantizer(model).compress()
```

File: docs/en_US/Release.md [hunk 2]
```diff
@@ -186,7 +186,7 @@
   * Run trial jobs on the GPU running non-NNI jobs
 * Kubeflow v1beta2 operator
   * Support Kubeflow TFJob/PyTorchJob v1beta2
-* [General NAS programming interface](AdvancedFeature/GeneralNasInterfaces.md)
+* General NAS programming interface
```
Comment (QuanluZhang) [inline docs/en_US/Release.md]: could you add the doc link in v0.8?

File: docs/en_US/Release.md [hunk 3]
```diff
@@ -299,7 +299,7 @@
 * Support [Metis tuner](Tuner/MetisTuner.md) as a new NNI tuner. Metis algorithm has been proofed to be well performed for **online** hyper-parameter tuning.
 * Support [ENAS customized tuner](https://github.com/countif/enas_nni), a tuner contributed by github community user, is an algorithm for neural network search, it could learn neural network architecture via reinforcement learning and serve a better performance than NAS.
 * Support [Curve fitting assessor](Assessor/CurvefittingAssessor.md) for early stop policy using learning curve extrapolation.
-* Advanced Support of [Weight Sharing](AdvancedFeature/AdvancedNas.md): Enable weight sharing for NAS tuners, currently through NFS.
+* Advanced Support of *Weight Sharing*: Enable weight sharing for NAS tuners, currently through NFS.
```
Comment (QuanluZhang) [inline docs/en_US/Release.md]: could you add the doc link in v0.5.0?

Comment (leckie-chn) [reply docs/en_US/Release.md]: * first, currently we only maintain documentation on `master` branch
* second, even v*.* branches are enabled on readthedocs, such cross branch links can only be implemented by `https://` links, which is not expected in our practice.

File: docs/en_US/Compressor/Quantizer.md (hunk 1 above)
Comment (chicm-ms) [reply docs/en_US/Compressor/Quantizer.md]: move the `python` up one line next to the quotes

**Pull request merged**